### DISCUSSION TITLE
Explicitly declaring implementation of protocol broken for dataclasses

### DISCUSSION THREAD

**Discussion by Germandrummer92** [python/typing#1121] [Q&A] [answered]
Hi,

Marking a dataclass as explicitly implementing a protocol (as specified in https://peps.python.org/pep-0544/#explicitly-declaring-implementation) seems to make the dataclass not instantiatable anymore. This seems like a bug to me, but I am also not sure where to open it? bugs.python.org? 

```python
from typing import Protocol
from dataclasses import dataclass

class SomeProtocol(Protocol):
    @property
    def some_value(self) -> str: ...

@dataclass
class SomeDataclasss(SomeProtocol):
    some_value: str

if __name__ == '__main__':
    a = SomeDataclasss(some_value="value") # this crashes with AttributeError: can't set attribute 'some_value'
```

**Comment by srittau** [comment, marked as answer]
Yes, this sounds like a clear bug to me. bugs.python.org is the right place indeed.

**Comment by Germandrummer92** [comment]
thanks, opened an issue: https://bugs.python.org/issue47237
